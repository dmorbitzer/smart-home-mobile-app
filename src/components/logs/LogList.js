import { ScrollView, StyleSheet } from "react-native";

import LogListItem from "./LogListItem";

export default function Services(props) {
  const buildLogs = (logs) => {
    return logs.map((element) => {
      if (props.filter && element.node.service.identifier !== props.filter) {
        return;
      }
      return (
        <LogListItem
          data={element.node.data}
          key={element.node.id}
          name={element.node.service.name}
          time={element.node.time}
          serviceType={element.node.service.identifier}
        />
      );
    });
  };
  return (
    <ScrollView style={styles.logList}>{buildLogs(props.logs)}</ScrollView>
  );
}

const styles = StyleSheet.create({
  logList: {
    height: "80%",
  },
});
