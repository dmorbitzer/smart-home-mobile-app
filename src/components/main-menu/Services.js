import { Box } from "@react-native-material/core";

import GetIcon from "../../api/GetIcon";
import IconButton from "../util/IconButton";

export default function Services(props) {
  const servicesList = props.data.map((element) => {
    if (element.node.active) {
      return (
        <IconButton
          key={element.node.name}
          title={element.node.name}
          type="primary"
          func={() => props.navigation.navigate(element.node.name)}
          icon={GetIcon(element.node.name)}
        />
      );
    }
  });
  return <Box h={100}>{servicesList}</Box>;
}
