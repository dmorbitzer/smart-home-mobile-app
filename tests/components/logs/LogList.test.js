import React from "react";
import renderer from "react-test-renderer";

import LogList from '../../../src/components/logs/LogList';
import Services from "../../../src/components/main-menu/Services";
jest.useFakeTimers();

const localData = {
    "data": {
        "logs": {
            "edges": [
                {
                    "node": {
                        "id": "/api/logs/1",
                        "service": {
                            "id": "/api/services/1",
                            "name": "Katzenfütterung",
                            "identifier": "cat_feeding"
                        },
                        "time": "2023-04-21T17:10:15+02:00",
                        "data": {
                            "cat": "Peter",
                            "foodType": "Trockenfutter"
                        }
                    }
                },
                {
                    "node": {
                        "id": "/api/logs/2",
                        "service": {
                            "id": "/api/services/1",
                            "name": "Katzenfütterung",
                            "identifier": "cat_feeding"
                        },
                        "time": "2023-08-21T17:10:20+02:00",
                        "data": {
                            "cat": "Paul",
                            "foodType": "Trockenfutter"
                        }
                    }
                },
            ]
        }
    }
}

test("renders Loading correctly", () => {
    const tree = renderer
        .create(<LogList logs={localData.data.logs.edges} filter={null} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
