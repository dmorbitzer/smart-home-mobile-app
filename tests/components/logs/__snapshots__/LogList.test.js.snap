// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading correctly 1`] = `
<RCTScrollView
  style={
    {
      "height": "80%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onLayout={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <View
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
            },
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginHorizontal": 16,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              selectable={false}
              style={
                [
                  {
                    "fontSize": 16,
                    "letterSpacing": 0.15,
                  },
                  {
                    "color": "#000000",
                  },
                  {
                    "color": "#212121",
                  },
                ]
              }
            >
              Katzenfütterung 2023-04-21T17:10:15+02:00
            </Text>
          </View>
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#666666",
                  "marginTop": 4,
                },
              ]
            }
          >
            Katze: Peter, Futter: Trockenfutter, Type: undefined
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#00000014",
              "height": 1,
              "marginEnd": undefined,
              "marginStart": 16,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onLayout={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <View
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
            },
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginHorizontal": 16,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              selectable={false}
              style={
                [
                  {
                    "fontSize": 16,
                    "letterSpacing": 0.15,
                  },
                  {
                    "color": "#000000",
                  },
                  {
                    "color": "#212121",
                  },
                ]
              }
            >
              Katzenfütterung 2023-08-21T17:10:20+02:00
            </Text>
          </View>
          <Text
            selectable={false}
            style={
              [
                {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                },
                {
                  "color": "#000000",
                },
                {
                  "color": "#666666",
                  "marginTop": 4,
                },
              ]
            }
          >
            Katze: Paul, Futter: Trockenfutter, Type: undefined
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#00000014",
              "height": 1,
              "marginEnd": undefined,
              "marginStart": 16,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
